client/scripts/controllers/chain/substrate/staking.ts

```diff
@@ -226,6 +257,84 @@ class SubstrateStaking implements StorageModule {
   public query(address: string): Observable<DeriveStakingQuery> {
     return this._Chain.query((api: ApiRx) => api.derive.staking.query(address));
   }
+  public get annualPercentRate(): Observable<ICommissionInfo> {
+    return this._Chain.api.pipe(
+      switchMap((api: ApiRx) => combineLatest(
+        of(api),
+        api.derive.staking.stashes(),
+        api.query.staking.currentEra(),
+        from(this._app.chainEvents.rewards()),
+        api.derive.staking.electedInfo()
+      )),
+      flatMap((
+        [api, allStashes, era, rewards, electedInfo]:
+        [ApiRx, AccountId[], EraIndex, IReward, DeriveStakingElected]
+      ) => {
+        const commission: ICommissionInfo = {};
+
+        electedInfo.info.forEach(({ accountId, validatorPrefs }) => {
+          const key = accountId.toString();
+          commission[key] = (validatorPrefs.commission.unwrap() || new BN(0)).toNumber() / 10_000_000;
+          return commission[key];
+        });
+        // Different runtimes call for different access to stakers: old vs. new
+        const stakersCall = (api.query.staking.stakers)
+          ? api.query.staking.stakers
+          : api.query.staking.erasStakers;
+        // Different staking functions call for different function arguments: old vs. new
+        const stakersCallArgs = (account) => (api.query.staking.stakers)
+          ? account
+          : [era.toString(), account];
+        return combineLatest(
+          stakersCall.multi(allStashes.map((elt) => stakersCallArgs(elt.toString()))),
+          of(allStashes),
+          of(rewards),
+          of(commission)
+        );
+      }),
+      auditTime(100),
+      map(([exposures, accounts, rewards, commissions ] :
+        [Exposure[], AccountId[], IReward, ICommissionInfo ]) => {
+        const n = 100000;
+        const validatorRewards: ICommissionInfo = {};
+
+        accounts.forEach((account, index) => {
+          const key = account.toString();
+          const exposure = exposures[index];
+          let stakeShare = 0;
+          let othersStake = (this._app.chain as Substrate).chain.coins(0);
+
+          exposures[index].others.forEach((indv) => {
+            othersStake = (this._app.chain as Substrate).chain.coins(othersStake.asBN.add(indv.value.toBn()));
+          });
+          othersStake = (this._app.chain as Substrate).chain.coins(othersStake.asBN.add(exposures[index].own.toBn()));
+          if (othersStake.gt((this._app.chain as Substrate).chain.coins(0))) {
+            stakeShare = exposures[index].own.toBn().muln(n).div(othersStake).toNumber() / n;
+          }
+
+          const totalStake = (this._app.chain as Substrate).chain.coins(exposure?.total.toBn())
+          || (this._app.chain as Substrate).chain.coins(0);
+          const totalReward: number = rewards.validators[key] || 0;
```

Comment:
The reward event doesn't always specify the validator as a map, it differs from Kusama and Edgeware.